include/bluetooth/gatt.h

```diff
@@ -1494,7 +1494,7 @@ struct bt_gatt_subscribe_params {
 	/** Subscribe value */
 	uint16_t value;
 	/** Subscription flags */
-	ATOMIC_DEFINE(flags, BT_GATT_SUBSCRIBE_NUM_FLAGS);
+	atomic_t flags[ATOMIC_BITS_TO_SZ(BT_GATT_SUBSCRIBE_NUM_FLAGS)];
```

Comment:
Not really happy with the inconsistency that this creates (not to mention it's less readable). There are many other places in Zephyr that use ATOMIC_DEFINE() inside structs. It just seems this is the only struct member with an associated doxygen documentation string. Would it be possible to fix this some other way, e.g. have doxygen/breathe look at the code only after pre-processor expansion?

Reply:
Thanks for the feedback @jhedberg.

I've found other possible solutions for this problem:

1. Use `EXPAND_AS_DEFINED` (ref. https://www.doxygen.nl/manual/config.html#cfg_expand_as_defined). This option requires `EXPAND_ONLY_PREDEF`, though. **After some tests I haven't been able to make it work...**
2. Add the `ATOMIC_DEFINE` macro in `PREDEFINED`, i.e. `ATOMIC_DEFINE(x, y)=atomic_t x[ATOMIC_BITS_TO_SZ(y)]`. This option requires to duplicate the actual macro definition in order to obtain correct Doxygen output. Result:

![image](https://user-images.githubusercontent.com/25011557/113491208-529b9080-94cf-11eb-9d0a-723354df373b.png)

For reference, the current output (incorrect):
![image](https://user-images.githubusercontent.com/25011557/113491179-21bb5b80-94cf-11eb-95e0-c00d5cd8cbc2.png)

Other usages found in `include` that are not rendered because field is hidden by a non-defined Kconfig option.

- include/net/ethernet.h:369:     ATOMIC_DEFINE(interfaces, NET_VLAN_MAX_COUNT);

Other usages found in `include` that do not render well in Doxygen but cause no warnings on Sphinx (maybe API is not exposed in the Sphinx side...?)

- include/net/net_if.h:435:       ATOMIC_DEFINE(flags, NET_IF_NUM_FLAGS);
- include/bluetooth/mesh/cdb.h:40:        ATOMIC_DEFINE(flags, BT_MESH_CDB_NODE_FLAG_COUNT);
- include/bluetooth/mesh/cdb.h:75:        ATOMIC_DEFINE(flags, BT_MESH_CDB_FLAG_COUNT);
- include/bluetooth/l2cap.h:102:  ATOMIC_DEFINE(status, BT_L2CAP_NUM_STATUS);

Reply:
The output of option 2 above looks quite good to me, in fact I'd say it's better than an explicit integer for the array size since what we're really interested in is how many bits the field can hold.

FWIW, I think the `ATOMIC_DEFINE()` macro itself might be a bit badly named since it's strictly about bitfields and not using atomic_t as a normal integer value. Perhaps `ATOMIC_BITFIELD_DEFINE()` might have been better. Anyway, that's a completely different discussion that doesn't have much to do with this PR.

Reply:
we may use `ATOMIC_BITMAP_SIZE()` instead of `ATOMIC_BITS_TO_SZ()`

Reply:
Yeah, that sounds good to me

Reply:
Btw, regarding the naming, Linux has a `DECLARE_BITMAP()` which is essentially exactly the same as our `ATOMIC_DEFINE()`